Bottom: 74c209e00a0d3f216f433c2cbf08dc3109397d02
Top:    d7ac26bf4b7c63103fe29c48cce8b16f56be4b0c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-01-11 13:38:49 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 08828b3..8db2f94 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1134,7 +1134,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
